extensions/data-workspace/src/services/workspaceService.ts

```diff
@@ -31,6 +31,21 @@ export class WorkspaceService implements IWorkspaceService {
 		this.getProjectsInWorkspace(undefined, true).catch(err => Logger.error(`Error initializing projects in workspace ${err}`));
 	}
 
+	get isProjectProviderAvailable(): boolean {
```

Comment:
You probably don't want to be doing this check every time. I'd suggest setting it on startup and then updating it whenever the extensions change.